api/client.go

```diff
@@ -183,6 +184,13 @@ func (c *Config) ReadEnvironment() error {
 	if v := os.Getenv(EnvVaultClientKey); v != "" {
 		envClientKey = v
 	}
+	if t := os.Getenv(EnvVaultClientTimeout); t != "" {
+		clientTimeout, err := strconv.ParseUint(t, 10, 32)
+		if err != nil {
+			return err
+		}
+		c.HttpClient.Timeout = time.Second * time.Duration(clientTimeout)
```

Comment:
The HTTP client might be nil here, and this is not the right place to create one. You should create a timeout variable in the Client struct, and assign it here (and in the set function). Then move the actual logic to NewClient after it guarantees that an HttpClient is created.